docs/recipes/custom-gatherer-puppeteer/searchable-gatherer.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* global document */
+
+const Gatherer = require('lighthouse').Gatherer;
+const Puppeteer = require('puppeteer-core');
+
+async function connect(driver) {
+  const browser = await Puppeteer.connect({
+    browserWSEndpoint: await driver.wsEndpoint(),
+    defaultViewport: null,
+  });
+  const {targetInfo} = await driver.sendCommand('Target.getTargetInfo');
+  const puppeteerTarget = (await browser.targets())
+    .find(target => target._targetId === targetInfo.targetId);
+  const page = await puppeteerTarget.page();
+  return {browser, page};
+}
+
+class TimeToSearchable extends Gatherer {
```

Comment:
I don't think the names / fake scenario here matters. I vote to just name these things `CustomGatherer` / `CustomAudit` ....

Reply:
yah works for me.

Reply:
well an **incorrect** fake name *definitely* isn't helpful ;)

here I agree with `CustomGatherer`/`CustomAudit`

Reply:
yeah I get the names as they are now are trash but thats because I am sabotaging them hehe